### PR TITLE
v24.3.2, v24.2.7, v24.1.9, v23.2.18 hotfix release notes

### PR DESCRIPTION
**Pull request by shannonbradshaw**

### CODE REVIEW

File: src/current/_includes/releases/v24.3/v24.3.2.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
+## v24.3.2
+
+Release Date: December 26, 2024
+
+{% include releases/new-release-downloads-docker-image.md release=include.release %}
+
+<h3 id="v24-3-2-sql-language-changes">SQL language changes</h3>
```
Comment (celiala) [inline src/current/_includes/releases/v24.3/v24.3.2.md:7]: this is the fix that's being introduced with the v24.3.2 (hot fix). from the 24.3 backport https://github.com/cockroachdb/cockroach/pull/137919:

> Release note (sql change): The legacy_varchar_typing session setting
has been added, which reverts the changes of https://github.com/cockroachdb/cockroach/pull/133037 that cause the
change in typing behavior described in https://github.com/cockroachdb/cockroach/issues/137837.

File: src/current/_includes/releases/v24.3/v24.3.2.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+- Added the `legacy_varchar_typing` session setting. When set to `on`, type-checking and overload resolution ignore the newly added `unpreferred` overload preference which causes a change in `VARCHAR` typing behavior (see [#133037][#133037]). The default value for `legacy_varchar_typing` is `on`. [#137943][#137943]
+
+<h3 id="v24-3-2-bug-fixes">Bug fixes</h3>
```
Comment (celiala) [inline src/current/_includes/releases/v24.3/v24.3.2.md:11]: This bug fix should be removed:

- this maps to https://github.com/cockroachdb/cockroach/pull/133037, which is the [buggy] bug fix that was introduced by v24.3.1, but is being reverted by the "hot fix" change (https://github.com/cockroachdb/cockroach/pull/137919 for 24.3)

**Review by celiala — COMMENTED** [review]
the hot fixes should be just 1 PR:
- the "SQL language change"

(the "Bug fix" should be removes, as this was a buggy bug fix introduced in the 23.2, 24.1, 24.2, 24.3 patches shipped last week)

**Comment by shannonbradshaw** [comment]
Wrapped all release notes for all four hotfix patches in this PR.

**Comment by shannonbradshaw** [comment]
@mgartner I would appreciate your review on this to ensure the language in the release note is correct. The note is identical across all four releases reflected in this PR, so if you review any one of the four .md files in the PR that will be sufficient.

**Comment by shannonbradshaw** [comment]
Preview here: https://676ca814a020d2ec34ececb5--cockroachdb-docs.netlify.app/docs/

File: src/current/_data/releases.yml [hunk 3]
```diff
@@ -7621,4 +7621,143 @@
...
+    linux_intel_fips: true
+    linux_arm_fips: false
+  docker:
+    docker_image: cockroachdb/cockroach
+    docker_arm: true
+    docker_arm_experimental: false
+    docker_arm_limited_access: false
+  source: true
+  previous_release: v24.3.1
+  cloud_only: true
```
Comment (celiala) [inline src/current/_data/releases.yml:7652]: let's remove these parameters for all patches:
- cloud_only
- cloud_only_message_short
- cloud_only_message

image

File: src/current/_includes/releases/v23.2/v23.2.18.md [hunk 4]
```diff
@@ -0,0 +1,14 @@
+## v23.2.18
+
+Release Date: December 26, 2024
+
+{% include releases/new-release-downloads-docker-image.md release=include.release %}
+
+<h3 id="v24-1-9-sql-language-changes">SQL language changes</h3>
+
+- Added the `legacy_varchar_typing` session setting. When set to `on`, type-checking and overload resolution for comparisons involving `VARCHAR` columns ignore the newly added `unpreferred` overload preference. The `unpreferred` preference introduced a change in `VARCHAR` typing behavior. For more information, see [#133037][#133037] and [#137837][#137837]. The `legacy_varchar_typing` session setting is `on` by default. [#137943][#137943]
+
```
Comment (celiala) [inline src/current/_includes/releases/v23.2/v23.2.18.md:10]: s/137943/137945/:

- v24.3.2 backport is https://github.com/cockroachdb/cockroach/pull/137943
- v23.2.18 backport is https://github.com/cockroachdb/cockroach/pull/137945

File: src/current/_includes/releases/v24.1/v24.1.9.md [hunk 5]
```diff
@@ -0,0 +1,14 @@
+## v24.1.9
+
+Release Date: December 26, 2024
+
+{% include releases/new-release-downloads-docker-image.md release=include.release %}
+
+<h3 id="v24-1-9-sql-language-changes">SQL language changes</h3>
+
+- Added the `legacy_varchar_typing` session setting. When set to `on`, type-checking and overload resolution for comparisons involving `VARCHAR` columns ignore the newly added `unpreferred` overload preference. The `unpreferred` preference introduced a change in `VARCHAR` typing behavior. For more information, see [#133037][#133037] and [#137837][#137837]. The `legacy_varchar_typing` session setting is `on` by default. [#137943][#137943]
+
```
Comment (celiala) [inline src/current/_includes/releases/v24.1/v24.1.9.md:10]: s/137943/137946/:

- https://github.com/cockroachdb/cockroach/pull/137946

File: src/current/_includes/releases/v24.2/v24.2.7.md [hunk 6]
```diff
@@ -0,0 +1,14 @@
+## v24.2.7
+
+Release Date: December 26, 2024
+
+{% include releases/new-release-downloads-docker-image.md release=include.release %}
+
+<h3 id="v24-1-9-sql-language-changes">SQL language changes</h3>
+
+- Added the `legacy_varchar_typing` session setting. When set to `on`, type-checking and overload resolution for comparisons involving `VARCHAR` columns ignore the newly added `unpreferred` overload preference. The `unpreferred` preference introduced a change in `VARCHAR` typing behavior. For more information, see [#133037][#133037] and [#137837][#137837]. The `legacy_varchar_typing` session setting is `on` by default. [#137943][#137943]
+
```
Comment (celiala) [inline src/current/_includes/releases/v24.2/v24.2.7.md:10]: s/137943/137944/:

- https://github.com/cockroachdb/cockroach/pull/137944

**Comment by shannonbradshaw** [comment]
@celiala thanks for the review and catches. Sorry to make you work so hard. I should have seen that the backport PR links were incorrect.

**Comment by shannonbradshaw** [comment]
@celiala I've addressed all your comments. PR is ready for a final review from you.

**Review by celiala — APPROVED** [review]
LGTM - thank you!

File: src/current/_includes/releases/v23.2/v23.2.18.md [hunk 7]
```diff
@@ -0,0 +1,14 @@
+## v23.2.18
+
+Release Date: December 26, 2024
+
+{% include releases/new-release-downloads-docker-image.md release=include.release %}
+
+<h3 id="v24-2-18-sql-language-changes">SQL language changes</h3>
+
+- Added the `legacy_varchar_typing` session setting. When set to `on`, type-checking and overload resolution for comparisons involving `VARCHAR` columns ignore the newly added `unpreferred` overload preference. The `unpreferred` preference introduced a change in `VARCHAR` typing behavior. For more information, see [#133037][#133037] and [#137837][#137837]. The `legacy_varchar_typing` session setting is `on` by default. [#137945][#137945]
```
Comment (mgartner) [inline src/current/_includes/releases/v23.2/v23.2.18.md:9]: I don't think this should mention "unpreferred" or "overloads"—those are implementation details. Those shouldn't have been in the release notes, apologies for that. How about something like this instead:

> Added the `legacy_varchar_typing` session setting. When set to `on`, type-checking comparisons involving `VARCHAR` columns behave as they did in all previous versions. When set to `off`, type-checking of these comparisons is more strict and queries that previously succeeded may now error with the message `unsupported comparison operator`. These errors can be fixed by added explicit type casts. The `legacy_varchar_typing` session setting is `on` by default.

**Comment by mgartner** [comment]
> @mgartner I would appreciate your review on this to ensure the language in the release note is correct. The note is identical across all four releases reflected in this PR, so if you review any one of the four .md files in the PR that will be sufficient.

@shannonbradshaw Just left one note about on the language.

**Comment by shannonbradshaw** [comment]
@mgartner will address in a follow up PR I'll put through now.

**Pull request merged**